chrome/background.js

```diff
@@ -12,82 +12,98 @@ chrome.webNavigation.onCompleted.addListener(async () => {
     );
 
   const [tab] = await getActiveTabs();
+  if (!tab) return;
   const currentHost = new URL(tab.url).hostname;
   const rules = elements[currentHost];
   if (!rules) return;
-
   for (const rule of rules) {
-    applyFunctions[`applyKind${rule.kind}`](tab, rule);
+    const {script, params} = applyFunctions[`applyKind${rule.kind}`](rule);
+      chrome.scripting.executeScript({
+        target: {tabId: tab.id}, 
+        args: [params],
+        func: script
+      });
   }
 });
 
-const applyKind1 = (tab, { selector, extraInfo }) => {
+const applyKind1 = ({ selector, extraInfo }) => {
   const toRemoveClasses = extraInfo
     .split(";")
     .map((klass) => `"${klass}"`)
     .join(",");
-
-  chrome.tabs.executeScript(tab.id, {
-    code: `document.querySelectorAll("${selector}").forEach(node => node.classList.remove(${toRemoveClasses}))`,
-  });
+  console.log(toRemoveClasses)
+  return {
+    script: ({selector, toRemoveClasses}) => document.querySelectorAll(selector).forEach(node => node.classList.remove(toRemoveClasses)),
+    params: {selector, toRemoveClasses}
+  }

```

Comment:
Is there any benefit on changing the `chrome.tabs.executeScript` to returning this object and executing it on the caller over 
executing `chrome.scripting.executeScript` here?

I'd say that if the reason is to avoid duplication we can move the `chrome.scripting.executeScript` to our own method and just call that one on each function kind